bin/tasks/report.js

```diff
@@ -61,30 +63,35 @@ export const generateReport = async (ctx) => {
     .property('buildUrl', build.webUrl)
     .property('storybookUrl', baseStorybookUrl(build.cachedUrl));
 
-  build.snapshots.forEach(({ status, spec, parameters }) => {
+  build.tests.forEach(({ status, result, spec, parameters }) => {
     const suffix = parameters.viewportIsDefault ? '' : ` [${parameters.viewport}px]`;
     const testCase = suite
       .testCase()
       .className(spec.component.name.replace(/[|/]/g, '.')) // transform story path to class path
       .name(`${spec.name}${suffix}`);
 
     switch (status) {
-      case 'SNAPSHOT_ERROR':
+      case 'FAILED':
         testCase.error('Server error while taking snapshot, please try again', status);
         break;
-      case 'SNAPSHOT_CAPTURE_ERROR':
+      case 'BROKEN':
         testCase.error('Snapshot is broken due to an error in your Storybook', status);
         break;
-      case 'SNAPSHOT_DENIED':
+      case 'DENIED':
         testCase.failure('Snapshot was denied by a user', status);
         break;
-      case 'SNAPSHOT_PENDING':
+      case 'PENDING':
         testCase.failure('Snapshot contains visual changes and must be reviewed', status);
         break;
-      case 'SNAPSHOT_NO_CAPTURE':
-        testCase.skipped();
-        break;
-      default:
+      default: {
+        switch (result) {
+          case 'SKIPPED':
+          case 'PRESERVED':
+            testCase.skipped();
+            break;
+          default:
+        }
```

Comment:
Seems a bit overkill to use a switch for something that could just be a single if statement.